src/Endpoint.cpp

```diff
@@ -194,14 +194,26 @@ namespace geopm
         while (m_continue_loop && agent == "") {
             agent = get_agent();
             std::this_thread::sleep_for(std::chrono::milliseconds(100));
-            if (geopm_time_since(&start) >= timeout) {
+            if (timeout >= 0 && geopm_time_since(&start) >= timeout) {
                 throw Exception("EndpointImp::" + std::string(__func__) +
                                 "(): timed out waiting for controller.",
                                 GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
             }
         }
     }
 
+    void EndpointImp::wait_for_agent_detach(double timeout)
+    {
+        std::string agent = get_agent();
+        double sample_age = 0;
+        std::vector<double> sample(m_num_sample);
+        while (m_continue_loop && agent != "" && sample_age < timeout) {
```

Comment:
here timeout == 0 -> do not wait, in wait_for_agent_attach timeout == 0 -> wait forever.  This seems a little confusing.

Reply:
The intention was that timeout==0 means do not wait.  I'll add a test for that.

Reply:
So the condition needs to check the value of agent in wait_for_agent_attach